server/bin/www

```diff
@@ -31,37 +32,78 @@ const server = http.createServer(app);
 
 const io = require("socket.io")(server);
 
+const wrap = (middleware) => (socket, next) =>
+  middleware(socket.request, {}, next);
+
+io.use(wrap(expressSession));
+
+io.use((socket, next) => {
+  const token = socket.handshake.auth.token;
+  if (token) {
+    jwt.verify(token, process.env.SESSION_SECRET, (err, decoded) => {
+      if (err) return next(new Error("not authorized"));
+      return next();
+    });
+  }
+
+  return next(new Error("not authorized"));
+});
+
 io.on("connection", (socket) => {
-  socket.on("go-online", (id) => {
-    if (!onlineUsers.includes(id)) {
-      onlineUsers.push(id);
+  console.log("socket server connected");
+
+  io.emit("socketId-collect");
+
+  socket.on("update-socketId", ({ userId, socketId }) => {
+    if (userId && socketId)
+      onlineUsers.set(userId, {
+        userId,
+        socketId,
```

Comment:
Nitpick: for onlineUsers, you don't need to save the userId in the object, since you already have it as the key.